backend/src/main/java/com/library/backend/controller/MealController.java

```diff
@@ -24,23 +24,50 @@ public ResponseEntity<List<Meal>> getAllMeals() {
     }
 
     @GetMapping("/{id}")
-     public ResponseEntity<Meal> getMealById(@PathVariable String id) {
+     public ResponseEntity<Meal> getMealById(@PathVariable String id) throws Exception {
         Meal meal = mealService.getMealById(id);
         return meal != null ? ResponseEntity.ok(meal) : ResponseEntity.notFound().build();
 
     }
 
     @GetMapping("/category/{category}")
     public ResponseEntity<List<Meal>> getMealsByCategory(@PathVariable String category) {
-        return ResponseEntity.ok(mealService.getMealsByCategory(category));
+        try {
+            return ResponseEntity.ok(mealService.getMealsByCategory(category));
+        } catch (Exception e) {
+            throw new  RuntimeException("Internal Server Error", e);
+
+        }
+    }
+    @ExceptionHandler(RuntimeException.class)
+    public ResponseEntity<String> handleRuntimeException(RuntimeException ex) {
+        // Hier kannst du die Nachricht und den Statuscode anpassen
+        return ResponseEntity.status(500).body(ex.getMessage());
```

Comment:
oftentimes we dont wan´t a 500, because the Problem is not our code but, for example, the wrong Input of the user